authors/apps/profiles/views.py

```diff
@@ -0,0 +1,36 @@
+from rest_framework import generics, status
+from rest_framework.response import Response
+from django.core.exceptions import ObjectDoesNotExist
+from .serializers import UserProfileSerializer
+from .models import UserProfile
+
+
+class ProfileView(generics.RetrieveUpdateAPIView):
+    '''
+    class view to update user profile
+    '''
+    serializer_class = UserProfileSerializer
+    queryset = UserProfile.objects.all()
+
+    def update(self, request, username, *args, **kwargs):
+        try:
+            user_instance = UserProfile.objects.select_related(
+                'user').get(user__username=username)
+            if user_instance.user.username != request.user.username:
+                return Response({'error': 'You are not allowed to edit or delete this object'}, status.HTTP_403_FORBIDDEN)
```

Comment:
Try having `{'error': 'You are not allowed to edit or delete this object'}` as a variable to make to easier to read